src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Tcp/ClientCredentialTypeTests.cs

```diff
@@ -122,4 +124,47 @@ public static void SameBinding_SecurityModeTransport_ClientCredentialTypeCertifi
             ScenarioTestHelpers.CloseCommunicationObjects(factory);
         }
     }
+
+    [Fact]
+    [ActiveIssue(458)]
+    [OuterLoop]
+    public static void TcpClientCredentialType_Certificate_EchoString()
+    {
+        string clientCertThumb = null;
+        EndpointAddress endpointAddress = null;
+        string testString = "Hello";
+        NetTcpBinding binding = null;
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy = null;
+
+        try
+        {
+            binding = new NetTcpBinding(SecurityMode.Transport);
+            binding.Security.Transport.ClientCredentialType = TcpClientCredentialType.Certificate;
+
+            endpointAddress = new EndpointAddress(new Uri(Endpoints.Tcp_ClientCredentialType_Certificate_Address),
+                new DnsEndpointIdentity("localhost")); // This Identity will be changed to use the Resource request that retruns the correct identity, pending Hong's merge.
+            clientCertThumb = BridgeClientCertificateManager.Thumbprint; // ClientCert as given by the Bridge
+
+            factory = new ChannelFactory<IWcfService>(binding, endpointAddress);
+            factory.Credentials.ServiceCertificate.Authentication.CertificateValidationMode = X509CertificateValidationMode.None;
+            factory.Credentials.ClientCertificate.SetCertificate(
+                StoreLocation.LocalMachine,
```

Comment:
Not sure how this would impact x-plat testing - I recall that we weren't able to access the LocalMachine store. Not blocking for this review.